### PR TITLE
Make use of mac() and ip() helpers.

### PR DESCRIPTION
**Pull request by inetuid**
Implements #31

### CODE REVIEW

**Review by mirceaulinic — APPROVED** [review]
This looks good @inetuid! Thanks!

File: napalm_eos/eos.py [hunk 1]
```diff
@@ -1277,7 +1277,7 @@ def _parse_per_peer_bgp_detail(peer_output):
                         unicode, item['last_event']))
                 item['remote_address'] = (
                     napalm_base.helpers.convert(
-                        unicode, item['remote_address']))
+                        napalm_base.helpers.ip, item['remote_address']))
```
Comment (mirceaulinic) [inline napalm_eos/eos.py]: should be `napalm_base.helpers.ip(item['remote_address']))`

File: napalm_eos/eos.py [hunk 2]
```diff
@@ -1292,10 +1292,10 @@ def _parse_per_peer_bgp_detail(peer_output):
                         unicode, item['routing_table']))
                 item['router_id'] = (
                     napalm_base.helpers.convert(
-                        unicode, item['router_id']))
+                        napalm_base.helpers.ip, item['router_id']))
```
Comment (mirceaulinic) [inline napalm_eos/eos.py]: should be `napalm_base.helpers.ip(item['router_id']))`

File: napalm_eos/eos.py [hunk 3]
```diff
@@ -1292,10 +1292,10 @@ def _parse_per_peer_bgp_detail(peer_output):
...
                 item['local_address'] = (
                     napalm_base.helpers.convert(
-                        unicode, item['local_address']))
+                        napalm_base.helpers.ip, item['local_address']))
```
Comment (mirceaulinic) [inline napalm_eos/eos.py]: should be `napalm_base.helpers.ip(item['local_address']))`

**Review by mirceaulinic — CHANGES_REQUESTED** [review]
Just noticed that you missed some details, please update @inetuid

File: napalm_eos/eos.py [hunk 4]
```diff
@@ -789,8 +789,8 @@ def get_arp_table(self):
             arp_table.append(
                 {
                     'interface' : interface,
-                    'mac'       : mac_format,
-                    'ip'        : ip,
+                    'mac'       : napalm_base.helpers.mac(mac_format),
```
Comment (mirceaulinic) [inline napalm_eos/eos.py]: Here you can also use `mac_raw` instead of `mac_format` and remove the unnecessary lines.

File: napalm_eos/eos.py [hunk 5]
```diff
@@ -957,7 +957,7 @@ def get_mac_address_table(self):
             moves       = mac_entry.get('moves', 0)
             mac_table.append(
                 {
-                    'mac'       : mac_format,
+                    'mac'       : napalm_base.helpers.mac(mac_format),
```
Comment (mirceaulinic) [inline napalm_eos/eos.py]: Same here, can use `mac_raw`

**Review by mirceaulinic — CHANGES_REQUESTED** [review]

File: napalm_eos/eos.py (hunk 1 above)
Comment (inetuid) [reply napalm_eos/eos.py]: convert() looks to call the first argument (napalm_base.helpers.ip) with the second argument (item['remote_address']):

```
def convert(to, who, default=u''):
    if who is None:
        return default
    try:
        return to(who)
    except:
        return default
```

So I don't think it's appropriate to pass napalm_base.helpers.ip(item['remote_address'])).

You could argue that the whole convert part is unnecessary, but I went for minimal changes.

Comment (mirceaulinic) [reply napalm_eos/eos.py]: Right, missed that `napalm_base.helpers.ip` is passed as argument for `convert`.

Comment (mirceaulinic) [reply napalm_eos/eos.py]: And yeah. you can get rid of the convert - not needed anymore

**Review by mirceaulinic — APPROVED** [review]

**Comment by inetuid** [comment]
Not sure if/what I need to do with this?

**Comment by mirceaulinic** [comment]
@inetuid Can you please rebase?

**Comment by dbarrosop** [comment]
If someone rebases this I will merge it and release a new version : )

**Comment by inetuid** [comment]
Assuming a rebase is fixing the merge (not updating the commit log) I made a change but it's now failing the unit tests. The tests pass if I run them locally on my development box but fail under Travis. Looks like the case difference in the mac() method is causing the problem.

**Comment by mirceaulinic** [comment]
@inetuid You did good. The new testing framework compares the expected data with the information returned by the getters - and seems to be case sensitive (e.g.: `{"expected": "08:00:27:10:c4:8f", "result": "08:00:27:10:C4:8F"}`). For the moment you can update the contents of the tests, converting all MAC addresses to upper in `expected_result.json` for the following tests:
- [test_get_arp_table](https://github.com/napalm-automation/napalm-eos/blob/develop/test/unit/mocked_data/test_get_arp_table/normal/expected_result.json) 
- [test_get_interfaces](https://github.com/napalm-automation/napalm-eos/blob/develop/test/unit/mocked_data/test_get_interfaces/normal/expected_result.json)

For traceroute, as the unresponsive hops are marked with \* (which obviously is not a valid IP address), you should do:

``` python
hop_addr = hop_details[4+probe_index*5]
ip_address = napalm_base.helpers.convert(napalm_base.helpers.ip, hop_addr, hop_addr)
```

Which returns the raw value (*) in case unable to convert to ip format.

File: napalm_eos/eos.py [hunk 6]
```diff
@@ -1190,7 +1188,7 @@ def traceroute(self, destination, source='', ttl=0, timeout=0):
             traceroute_result['success'][hop_index] = {'probes': {}}
             for probe_index in range(probes):
                 host_name = hop_details[3+probe_index*5]
-                ip_address = hop_details[4+probe_index*5]
+                ip_address = napalm_base.helpers.ip(hop_details[4+probe_index*5])
```
Comment (mirceaulinic) [inline napalm_eos/eos.py]: Replace with this @inetuid 

``` python
hop_addr = hop_details[4+probe_index*5]
ip_address = napalm_base.helpers.convert(napalm_base.helpers.ip, hop_addr, hop_addr)
```

**Review by mirceaulinic — CHANGES_REQUESTED** [review]

**Comment by inetuid** [comment]
Changes made.

**Pull request merged**